tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/Session.java

```diff
@@ -715,7 +719,7 @@ Graph graph() {
   private int numActiveRuns;
 
   private final boolean autoInit;
-  private Set<Operation> ranInits = new LinkedHashSet<>();
+  private Set<Operation> ranInits = Collections.synchronizedSet(new LinkedHashSet<>());
```

Comment:
Why is this a synchronized set? It's either empty, or set to `graph.initializers()` (which is an unmodifiable view on a synchronized set). As far as I can tell the object allocated here is never mutated.